Packages/com.unity.inputsystem/InputSystem/Editor/PropertyDrawers/AssetSearchProviders.cs

```diff
@@ -0,0 +1,89 @@
+    {
+        // Note that if this method is annotated with [SearchItemProvider] we would register the provider to
+        // the SearchWindow, we currently skip this since we are mainly interested in using SearchItemProvider as
+        // a picker.
+        private static SearchProvider CreateAssetSearchProvider(string id, string displayName,
+            Func<IEnumerable<Object>> assetProvider, Texture2D thumbnail = null)
+        {
+            return new SearchProvider(id, displayName)
+            {
+                priority = 25,
+                toObject = GetObject,
+                fetchItems = (context, items, provider) => Search(context, provider, assetProvider, thumbnail),
+            };
+        }
+
+        internal static SearchProvider CreateInputActionReferenceSearchProvider()
+        {
+            // Match icon used for sub-assets from importer, if null will use cached icon from asset database instead.
+            // Note that .inputactions has one icon and sub-assets (actions) have another in the importer.
+            return CreateAssetSearchProvider(
+                "InputActionReferenceSearchProvider",
+                "Project-Wide Input Actions",
+                GetInputActionReferenceAssets,
+                (Texture2D)EditorGUIUtility.Load(InputActionImporter.kActionIcon));
+        }
+
+        private static IEnumerable<Object> GetInputActionReferenceAssets()
+        {
+            // Extract actions from actions maps from ProjectWideActionAsset.
+            // Note that we cannot do something like:
+            //  return AssetDatabase.LoadAllAssetsAtPath(ProjectWideActionsAsset.kAssetPath)
+            //      .Where((asset) => (asset is InputActionReference));
+            // Since that would return an outdated asset that is out of sync with ProjectWideActionsAsset.
+            // TODO Understand why this is a problem and update this comment
+            var asset = ProjectWideActionsAsset.GetOrCreate();
+            return (from actionMap in asset.actionMaps from action in actionMap.actions select InputActionReference.Create(action)).ToList();
+        }
+
+        static IEnumerable<SearchItem> Search(
+            SearchContext context, SearchProvider provider, Func<IEnumerable<Object>> assetProvider, Texture2D thumbnail)
+        {
+            foreach (var asset in assetProvider())
+            {
+                var label = asset.name;
+                if (!label.Contains(context.searchText, System.StringComparison.InvariantCultureIgnoreCase))
+                    continue; // Ignore: not matching search text filter
+
+                yield return provider.CreateItem(context,
+                    asset.GetInstanceID().ToString(),
+                    label,
+                    $"{AssetDatabase.GetAssetPath(asset)} ({label})",
+                    (thumbnail == null) ? AssetDatabase.GetCachedIcon(ProjectWideActionsAsset.kAssetPath) as Texture2D : thumbnail,
```

Comment:
I like that we have a new icon to indicate the project wide ones

These icons seem to update at I'm viewing the UI which is a bit distracting. 
(Would be nice if it was at least blank before updating to reduce distraction.)

I think this is potentially a UI issue outside the scope of this PR though.

Reply:
Icons I'm seeing - opportunity to revise the icon for project wide input actions (currently seems to be a magnifying glass)

![image](https://github.com/Unity-Technologies/InputSystem/assets/33493311/0f1ef8f2-5a69-4b70-918d-66c0b1123ef4)

Reply:
Thing is we do not have any new icons, I just used the one registered with asset database. The icon (blue+orange) is the .inputactions icon registered by InputActionImporter. This should really be the way to fetch it (the default value handling I have in the code). However the "Flash icon" is what is I currently set for the Project-wide in this search. That is currently used with .inputactions sub-assets (InputActions), but it could be debated what to use. I see the choice of icon as minor for this PR and could be addressed separately. However I propose here to differentiate assets and project-wide for increased visual cue.

Reply:
The magnifying glass is a default icon from the SearchProvider

Reply:
@lyndon-unity Do you see Input Action Icon (Flash) in Project-Wide Input Actions search group?

Reply:
> I like that we have a new icon to indicate the project wide ones
> 
> These icons seem to update at I'm viewing the UI which is a bit distracting. (Would be nice if it was at least blank before updating to reduce distraction.)
> 
> I think this is potentially a UI issue outside the scope of this PR though.

Not sure was this was referring to? I need to understand this to address it I believe